src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Test
+    void addUser_LoginLessThanMinLength_ThrowsException() {
+        User user = new User();
+        user.setLogin("Bob");
+        user.setPassword("qwerty1234");
+        user.setAge(25);
+        assertThrows(RegistrationException.class, () -> REGISTRATION_SERVICE.register(user));
+    }
+
+    @Test
+    void addUser_PasswordLessThanMinLength_ThrowsException() {
+        User user = new User();
+        user.setLogin("Bob1256");
+        user.setPassword("qwert");
+        user.setAge(25);
+        assertThrows(RegistrationException.class, () -> REGISTRATION_SERVICE.register(user));
+    }
+
+    @Test
+    void addUser_Underage_ThrowsException() {
+        User user = new User();
+        user.setLogin("Alice");
+        user.setPassword("qwerty1234");
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> REGISTRATION_SERVICE.register(user));
+    }
+
+    @Test
+    void addUser_WithValidDetails_Success() {
+        User user = new User();
+        user.setPassword("qwerty123");
+        user.setLogin("Alex1976");
+        user.setAge(27);
+        User registeredUser = REGISTRATION_SERVICE.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void addUser_InvalidAge_ThrowsException() {
+        User user = new User();
+        user.setLogin("Bob1256");
+        user.setPassword("password");
+        user.setAge((int) (Integer.MAX_VALUE + 1L));
+        assertThrows(RegistrationException.class, () -> REGISTRATION_SERVICE.register(user));
+    }
+
+    @Test
+    void testGetUserByLogin_Success() {
+        User user = new User();
+        user.setLogin("abcdr22");
+        user.setPassword("jjjjjjj");
+        user.setAge(19);
+        REGISTRATION_SERVICE.register(user);
+        User retrievedUser = STORAGE_DAO.get(user.getLogin());
+        assertEquals(user, retrievedUser);
+    }
+
+    @Test
+    void testGetUserByNonexistentLogin_ReturnsNull() {
+        User user = new User();
+        user.setLogin("Alice123");
+        user.setPassword("qwerty1234");
+        user.setAge(19);
+        REGISTRATION_SERVICE.register(user);
+        User retrievedUser = STORAGE_DAO.get("NonexistentLogin");
+        assertNull(retrievedUser);
+    }
+
+    @Test
+    void testGetUserWithoutGetterMethod_Success() {
+        User user = new User();
+        user.setLogin("Alice123");
+        user.setPassword("qwerty1234");
+        user.setAge(19);
+        REGISTRATION_SERVICE.register(user);
+        User retrievedUser = STORAGE_DAO.get("Alice123");
+        assertEquals(user, retrievedUser);
+    }
+
+    @Test
+    void testGetUserAfterDirectStorageAdd_Success() {
+        User user = new User();
+        user.setLogin("Alice123");
+        user.setPassword("qwerty1234");
+        user.setAge(19);
+        Storage.people.add(user);
+        User retrievedUser = STORAGE_DAO.get(user.getLogin());
+        assertEquals(user, retrievedUser);
+    }
```

Comment:
The test suite does not cover all possible edge cases for user parameters, such as a null user object, a negative age, or a user with an age of exactly 18 (which is an edge case for the minimum age). Make sure to include these tests to ensure comprehensive coverage.